### ISSUE TITLE
Consider contrib/ directory for non-glyph copyright stuff

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7616]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)| @moshez reported|
|-|-|
|Trac ID|trac#350|
|Type|enhancement|
|Created|2003-10-19 13:19:53Z|

<details><summary>Searchable metadata</summary>

```
trac-id__350 350
type__enhancement enhancement
reporter__moshez moshez
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1066569593000000 1066569593000000
changetime__1468845978329518 1468845978329518
version__ 
owner__etrepum etrepum
cc__glyph cc__etrepum cc__moshez
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Everything under Twisted/ should be licensed under the GPL.
the WebServicesTool is licensed under MIT. While the licenses are compatible, 
this just creates a mess. You are free, of course, to package WebServicesTool
separately, in addition, and license it however you wish.

Thanks.
doc/examples/Cocoa/WebServicesTool/LICENSE.txt
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="etrepum's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|etrepum commented|
|-|-|

```
#!html
<pre>
I didn't write the original, I can ask bbum if he'll relicense it if necessary.  The licenses are 
compatible, it's a documentation example, and it wouldn't bother anyone if you excluded it from 
the Debian distribution.  This issue isn't really worth my time.  It's up to glyph.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
I'm not going to generate a separate source tarball, and the debian 
distribution includes the source.
It will bother me greatly if it is AT ALL under Twisted/
If you can't relicense, neither can I.
If you can, please do so.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
I've removed it for the time being.
Please re-add it if and when you have time to properly get copyright
assignments.
I've renamed the bug
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
Let's exclude it anywhere it needs to be excluded, add it to the list of 
exceptions in the README, etc.  I don't think we should get rid of it or 
move it to another project.  Platform support is sort of inherently a 
licensing problem anyway - I mean, is it *really* OK to include code 
that depends on win32 APIs into the debian distribution?  who can say.

Bob, this is a pretty mild case, but *PLEASE* alert me in advance if 
you're going to check in code that you don't legally have the right to. 
  It could be a much worse mess.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="etrepum's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|etrepum commented|
|-|-|

```
#!html
<pre>
Where, specifically, do these exceptions need to go?  Supposing of course that I don't hear back 
from bbum, or he denies my request for permission to relicense.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Well, assuming we want exceptions [and I'm seriously against having examples
which can't be used to then base a HOWTO on, etc.], at the minimum the steps
should be:
* mention in the README
* put the license in the relevant place
* notify me so I can update debian/copyright
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
With the project getting to the size it is, and especially given our 
ambitious goals for platform support, we are likely to run into more of 
these issues.

We probably have to come up with a more graceful solution than these 3 
steps: perhaps we can have a contrib/ directory that's in the same CVS 
but distributed as a separate tarball for debian sources and the 
"average" download, but included in a "sumo" tarball or available 
separately?

I hate this kind of thing, especially for licensing reasons, but I'd 
rather have two cleanly separated licensing messes than a README for the 
base framework that has 45 copyrights tacked on near the bottom so we 
can support Mac OS and RISCOS and whatever it is that runs on the atari ST.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Please.
The issue was never Cocoa support. The issue was a Cocoa example someone
else wrote. I'm not sure why you forsee more problems.
[The last time we had this problem was because we used half-ass infrastructure
which was not in Python. We ended up rewriting all of it because it was
not good enough anyway. Not because of platform support, which is a red
herring.]

Re: win32: Debian supports win32 just fine through the Wine package.
And if anyone sends bugs for Twisted-win32/Wine problems, they will
be dealt with by me.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="etrepum's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|etrepum commented|
|-|-|

```
#!html
<pre>
Ok, example is out.  bbum doesn't like LGPL and it's not compatible with his employment at Apple.

The example will be distributed separately, either with PyObjC or by myself elesewhere.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
For future clarification: the real issue here is that the copyright hasn't been assigned to Glyph, not 
(just) that the license is not LGPL. Correct?

Those two different issues seem to have been somewhat commingled in the comments here.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
Twisted no longer requires copyright assignment to glyph for inclusion, thus this is no longer a 
problem.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_13"></a>|GitHub commented|
|-|-|

In changeset f1328d4c427648226728b587255c9f2edfeef2f5
```
#!CommitTicketReference repository="" revision="f1328d4c427648226728b587255c9f2edfeef2f5"
Merge pull request #350 from twisted/8621-rodrigc-spread

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8621

In twisted/spread, fix imports on Python 3
```
